crates/primitives/starknet/src/traits/hash.rs

```diff
@@ -11,24 +11,15 @@ pub trait HasherT {
     /// * `data` - The data to hash.
     /// # Returns
     /// The hash of the data.
-    fn hash(&self, data: &[u8]) -> Felt252Wrapper;
+    fn hash_bytes(&self, data: &[u8]) -> Felt252Wrapper;
 
     // Hashes the given data.
     /// # Arguments
     /// * `data` - The data to hash.
     /// # Returns
     /// The hash of the data.
-    fn hash_elements(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper;
-}
-
-/// A trait for default hashing instance.
-pub trait DefaultHasher {
-    /// Get Hasher default instance.
-    fn hasher() -> Self;
-}
+    fn compute_hash_on_wrappers(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper;
```

Comment:
I feel like `compute_hash_on_wrappers` is redundant since it's the same underlying structure. Let's keep it simple here and only keep the wrapper version.

Reply:
I somewhat need guidance on that ; it's true `compute_hash_on_wrappers` and `compute_hash_on_elements` essentially do the same but the latter doesn't need to be passed `&self`into the parameters.

We could remove it and just use the former but I think it would require refactoring [primitives/starknet/crypto/commitment](https://github.com/keep-starknet-strange/madara/blob/main/crates/primitives/starknet/src/crypto/commitment/mod.rs) and everywhere its functions are used. Basically, if using `compute_hash_on_wrappers`, would need an instance of a `HasherT` to be passed into every function where the compute is used.

Reply:
I think this would make a lot of sense, anyway I changed every `FieldElement` to `Felt252Wrapper` in the global state root [PR](https://github.com/keep-starknet-strange/madara/pull/667) so we can do it. It just makes sense to unify everything and since the repo clearly tends to use the wrapper everywhere, we should also do it here imho.

Reply:
What if we replace `compute_hash_on_elements` by a similarly named method that uses `Felt252Wrapper` but unlike `compute_hash_on_wrappers` does not need to be passed `&self`? In some places, it just seems to make more sense that the generic parameter `T: HasherT` is not instantiated and we just directly call the appropriate function.

Reply:
Sounds like a great idea, please do :) Overall I think we should focus on making starknet-rs core no_std compatible would also reduce the complexity of this PR and just make things more straightforward.

Reply:
Yup, I'll be tackling the starknet-core `no-std` support this week